### PR TITLE
Add snippet to create a host

### PR DESCRIPTION
**Pull request by maximiliankolb**
* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [x] Foreman 3.8/Katello 4.10
* [x] Foreman 3.7/Katello 4.9 (planned Satellite 6.14)

### CODE REVIEW

File: guides/common/assembly_provisioning-cloud-instances-azure.adoc [hunk 1]
```diff
@@ -12,7 +12,9 @@ include::modules/proc_adding-images-to-server.adoc[leveloffset=+1]
 
 include::modules/proc_adding-microsoft-azure-resource-manager-details-to-a-compute-profile.adoc[leveloffset=+1]
 
+:creating-image-only-hosts:
 include::modules/proc_creating-image-only-hosts.adoc[leveloffset=+1]
+:!creating-image-only-hosts:
```
[the remark this answers is not in the document]
Comment (maximiliankolb) [reply guides/common/assembly_provisioning-cloud-instances-azure.adoc:17]: Yes it is: https://docs.asciidoctor.org/asciidoc/latest/attributes/unset-attributes/

**Comment by maximiliankolb** [comment]
Building this was challenging, I suspect so is the review.

| my PR | current nightly |
|--------|--------|
| [6.3](https://theforeman-foreman-documentation-preview-pr-2492.surge.sh/nightly/Provisioning_Hosts/index-katello.html#Creating_Hosts_with_Unattended_Provisioning_provisioning) | [6.3](https://docs.theforeman.org/nightly/Provisioning_Hosts/index-katello.html#Creating_Hosts_with_Unattended_Provisioning_provisioning) |
| [6.4](https://theforeman-foreman-documentation-preview-pr-2492.surge.sh/nightly/Provisioning_Hosts/index-katello.html#Creating_Hosts_with_PXE_less_Provisioning_provisioning) | [6.4](https://docs.theforeman.org/nightly/Provisioning_Hosts/index-katello.html#Creating_Hosts_with_PXE_less_Provisioning_provisioning) |
| [6.5](https://theforeman-foreman-documentation-preview-pr-2492.surge.sh/nightly/Provisioning_Hosts/index-katello.html#Creating_Hosts_with_UEFI_HTTP_Boot_Provisioning_provisioning) | [6.5](https://docs.theforeman.org/nightly/Provisioning_Hosts/index-katello.html#Creating_Hosts_with_UEFI_HTTP_Boot_Provisioning_provisioning) |
| [9.5](https://theforeman-foreman-documentation-preview-pr-2492.surge.sh/nightly/Provisioning_Hosts/index-katello.html#creating-network-or-image-based-hosts_kvm-provisioning) | [9.5](https://docs.theforeman.org/nightly/Provisioning_Hosts/index-katello.html#creating-network-or-image-based-hosts_kvm-provisioning) |
| Cell | Cell | 
| Cell | Cell | 
| Cell | Cell |

File: guides/common/assembly_using-pxe-to-provision-hosts.adoc [hunk 2]
```diff
@@ -4,10 +4,16 @@ include::modules/con_prerequisites-for-bare-metal-provisioning.adoc[leveloffset=
 
 include::modules/proc_configuring-the-security-token-validity-duration.adoc[leveloffset=+1]
 
+:creating-hosts-with-unattended-provisioning:
```
Comment (maximiliankolb) [inline guides/common/assembly_using-pxe-to-provision-hosts.adoc:7]: maybe introduce "pxe-provisioning" attribute similar to proxmox/vmware?

Comment (Lennonka) [reply guides/common/assembly_using-pxe-to-provision-hosts.adoc:7]: Isn't PXE common to all bare metal (with PXE) and VM provisioning?
In other words, I don't think that this assembly follows the same principle as proxmox/vmware...
I wouldn't introduce an attribute for this.

File: Dockerfile
Comment (Lennonka) [inline Dockerfile:1]: I think it's bad form making Docker changes in the same PR as content changes :)) Could you please extract them to a different PR?

File: guides/common/modules/snip_creating-a-host-host-tab.adoc
Comment (Lennonka) [inline guides/common/modules/snip_creating-a-host-host-tab.adoc:1]: Can we call this file `snip_steps-create-a-host-tab-host`? (At least with the `steps-` infix.)

File: guides/common/modules/snip_creating-a-host-interfaces-tab.adoc
Comment (Lennonka) [inline guides/common/modules/snip_creating-a-host-interfaces-tab.adoc:1]: Can we call this file `snip_steps-create-a-host-tab-interfaces`? (At least with the `steps-` infix.)

File: guides/common/modules/snip_creating-a-host-interfaces-tab.adoc [hunk 3]
```diff
@@ -0,0 +1,32 @@
+. Click the *Interface* tab, and on the interface of the host, click *Edit*.
```
Comment (Lennonka) [inline guides/common/modules/snip_creating-a-host-interfaces-tab.adoc:1]:
```suggestion
. Click the *Interfaces* tab, and on the interface of the host, click *Edit*.
```

File: guides/common/modules/snip_creating-a-host-interfaces-tab.adoc [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+. Verify that the fields are populated with values.
+Note in particular:
++
+* The *Name* from the *Host* tab becomes the *DNS name*.
+* {Project} automatically assigns an IP address for the new host.
++
+ifdef::creating-hosts-with-unattended-provisioning,uefi-http-boot-provisioning,creating-hosts-with-pxeless-provisioning[]
```
Comment (Lennonka) [inline guides/common/modules/snip_creating-a-host-interfaces-tab.adoc:8]:
```suggestion
ifdef::unattended-provisioning,uefi-http-boot-provisioning,pxeless-provisioning[]
```
Is there a reason to not make these attribute names shorter?

File: guides/common/modules/snip_creating-a-host-interfaces-tab.adoc
Comment (Lennonka) [inline guides/common/modules/snip_creating-a-host-interfaces-tab.adoc:1]: The order of the UI fields should follow the order in the WebUI:
image

**Review by Lennonka — CHANGES_REQUESTED** [review]
A couple of suggestions. I haven't checked yet, whether the snippet should be included somewhere else, I'll let you know.

**Comment by Lennonka** [comment]
I confirm you've replaced everything in _Provisioning Hosts_.

We also have ["Creating a Host" in _Managing Hosts_](https://docs.theforeman.org/nightly/Managing_Hosts/index-katello.html#Creating_a_Host_managing-hosts), however:

1. I don't think it should be in that guide, but that's out of scope of this PR.
2. Not sure if the snippets should replace parts of that as well.

**Comment by Lennonka** [comment]
I think we should cherry-pick it as far as possible, otherwise all future cherry-picks are going to be a huge pain, but we must verify if there are any changes in the WebUI.

**Review by Lennonka — CHANGES_REQUESTED** [review]
Looks good, just a couple of details.

@maximiliankolb Can we cherry-pick this as I mentioned above?

File: guides/common/modules/snip_steps-create-a-host-tab-host.adoc [hunk 5]
```diff
@@ -0,0 +1,4 @@
+. In the {ProjectWebUI}, navigate to *Hosts* > *Create Host*.
+. In the *Name* field, enter a name for the host.
+. Click the *Organization* and *Location* tabs and change the context to match your requirements.
```
[the remark this answers is not in the document]
Comment (maximiliankolb) [reply guides/common/modules/snip_steps-create-a-host-tab-host.adoc:3]: Yes and Yes.

File: guides/common/modules/snip_steps-create-a-host-tab-host.adoc [hunk 6]
```diff
@@ -0,0 +1,4 @@
...
+. From the *Host Group* list, select a host group that you want to use to populate the form.
```
[the remark this answers is not in the document]
Comment (maximiliankolb) [reply guides/common/modules/snip_steps-create-a-host-tab-host.adoc:4]: Applied.

**Review by maximiliankolb — COMMENTED** [review]
Applied your feedback. I am unsure how much work it is to cherry-pick all the way down to 3.1, but we can certainly try!

**Review by Lennonka — APPROVED** [review]
Alrighty, I'll try to cherry-pick it as far as possible and let you know :)

**Comment by maximiliankolb** [comment]
Merged to master and started cherry-picking: :rocket:  
   d50820ed8..fdcac1656  3.8 -> 3.8
   c986bde5c..0ef2c2a88  3.7 -> 3.7

I will continue one by one because I have a feeling that things might break after 3.7 has been built. Will keep you posted @Lennonka

File: guides/common/assembly_provisioning-cloud-instances-azure.adoc [hunk 7]
```diff
@@ -1,4 +1,7 @@
-ifdef::context[:parent-context: {context}]
+:parent-context: {context}
+:context: azure-provisioning
```
Comment (Lennonka) [inline guides/common/assembly_provisioning-cloud-instances-azure.adoc:2]: @maximiliankolb Max, we didn't realize that changing the context will mess up our links! I think we shouldn't change the context at all in these places and we should only define the additional attributes such as `:azure-provisioning:`, which will not mess up the context in links. We don't use the context attribute to condition the provisioning content anyway.

Comment (Lennonka) [reply guides/common/assembly_provisioning-cloud-instances-azure.adoc:2]: I will file a PR to fix that.
